worker/provisioner/provisioner_task.go

```diff
@@ -1028,7 +1039,7 @@ func (task *provisionerTask) startMachine(
 		} else if attemptsLeft <= 0 {
 			// Set the state to error, so the machine will be skipped
 			// next time until the error is resolved.
-			return task.setErrorStatus("cannot start instance for machine %q: %v", machine, err)
+			return task.setErrorStatusAndRemoveMachineAZ("cannot start instance for machine %q: %v", machine, err)
```

Comment:
I was thinking this would cause the provisioner to stop, but the error doesn't propagate that far. So just return an error, and let it be logged by startMachines (which will also remove from the map)

Reply:
the code has changed, so that error propagated up would now cause the provisioner to stop.